File: code/b22163.pas
program bai3denghean;
uses math;
var
n, x, y : longint;
c : Array[1..100000] of longint;
dem, i,j : longint;
begin
    assign(input,'trungthuong.inp'); reset(input);
    assign(output,'trungthuong.out'); rewrite(output);
    
    readln(n);
    for i:= 1 to n do read(c[i]);
    for i := 1 to (n - 1) do
    begin
        y := c[i];
        x := c[i];
        for j := i + 1 to n do 
        begin
            if x > c[j] then x := c[j];
            if y < c[j] then y := c[j];
            if (x= min(c[i], c[j])) and (y = max(c[i], c[j]))
            then dem := dem +1;
        end;
    end;
    writeln(dem);
end.

File: code/b22163.py
import sys
sys.stdin = open("trungthuong.inp","r")
sys.stdout = open("trungthuong.out","w")
n = int(input())
c = [int(x) for x in input().split()]
dem = 0
for i in range (n - 1):
    x = y = c[i]
    for j in range (i + 1,  n):
        if x > c[j]: x = c[j]
        if y < c[j]: y = c[j]
        if (x== min(c[i], c[j])) and (y == max(c[i], c[j])):
            dem += 1
print(dem)
